store/src/java/com/zimbra/cs/util/ProxyConfGen.java

```diff
@@ -1196,10 +1196,41 @@ public String format(Object o) {
     }
 }
 
+/**
+ * ListenAddressesVar
+ * This class is intended to produce strings that are embedded inside the 'nginx.conf.web.https.default' template
+ * It is placed inside the strict server_name enforcing server block.
+ */
+class ListenAddressesVar extends ProxyConfVar {
+
+    public ListenAddressesVar(Set<String> addresses) {
+        super("listen.:addresses",
+                null, // this is a fake attribute
+                addresses,
+                ProxyConfValueType.CUSTOM,
+                ProxyConfOverride.CUSTOM,
+                "List of ip addresses nginx needs to listen to catch all unknown server names");
+    }
+
+    @Override
+    public String format(Object o) throws ProxyConfException {
+       Set<String> addresses = (Set<String>)o;
+        if (addresses.size() == 0) {
+            return "";
+        }
+        StringBuilder sb = new StringBuilder();
+        for (String addr: addresses) {
+            sb.append(String.format("${web.strict.servername}    listen                  %s:${web.https.port} default_server;\n", addr));
+        }
+        sb.setLength(sb.length() - 1); //trim the last space
```

Comment:
That's going to be a newline, not a space isn't it?

Reply:
I have to be honest I copy and pasted this section and didn't actually pay attention to the comment.
It does appear to format properly in the file without adding an extra newline to the last item.

Reply:
I will update the comment accordingly.